lib/browser-auto-tracking.js

```diff
@@ -161,41 +160,52 @@ export function initAutoTrackingCore(lib) {
           time_on_page: allTimeOnSiteS > 0 ? allTimeOnSiteS : getSecondsSinceDate(now),
           time_on_page_ms: allTimeOnSiteMS > 0 ? allTimeOnSiteMS : getMiliSecondsSinceDate(now)
         },
-
+    
         ip_address,
         geo: { /* Enriched */ },
-
+    
         user_agent: '${keen.user_agent}',
         tech: {
           profile: browserProfile
           /* Enriched */
         },
-
+    
         url: {
           full: window ? window.location.href : '',
           info: { /* Enriched */ }
         },
-
+    
         referrer: {
           initial: initialReferrer,
           full: document ? document.referrer : '',
           info: { /* Enriched */ }
         },
-
+    
         time: {
           local: { /* Enriched */ },
           utc: { /* Enriched */ }
         },
-
+    
         keen: {
           timestamp: new Date().toISOString(),
           addons,
         }
       };
+    }
+
+    client.extendEvents(function() {
+      const browserProfile = helpers.getBrowserProfile();
+
+      if (browserProfile instanceof Promise) {
+        return browserProfile
+        .then(data => getExtendedObject(data));
```

Comment:
this one still returns a Promise, so you'll end up with JSON field value like:
```javascript
{
  battery: 'Promise {<pending>}'
}
```

To solve this issue we need to wait for all Promises to be resolved, before sending any event to the api.
This will require more logic around client.recordEvent

Reply:
https://jsfiddle.net/pm2astjy/

Reply:
let's just wrap the recording functions with a promise like:
```
...
first collect all of the data
...
someFuncPreparingTheData.then(dataWithBatteryStatusEtc => {

then add listeners eg

    if (options.recordClicks === true) {
      utils.listener('a, a *, button').on('click', function(e) {
     recordEvent()....
}
...
...